File: Chronir-Android/core/model/src/main/java/com/chronir/model/Alarm.kt
package com.chronir.model

import java.time.Instant
import java.time.LocalTime
import java.util.UUID

enum class CycleType {
    ONE_TIME,
    WEEKLY,
    MONTHLY_DATE,
    MONTHLY_RELATIVE,
    ANNUAL,
    CUSTOM_DAYS;

    val displayName: String
        get() = when (this) {
            ONE_TIME -> "One-Time"
            WEEKLY -> "Weekly"
            MONTHLY_DATE -> "Monthly"
            MONTHLY_RELATIVE -> "Monthly"
            ANNUAL -> "Annual"
            CUSTOM_DAYS -> "Custom"
        }
}

data class Alarm(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val cycleType: CycleType = CycleType.WEEKLY,
    val timeOfDay: LocalTime = LocalTime.of(8, 0),
    val schedule: Schedule = Schedule.Weekly(daysOfWeek = listOf(1), interval = 1),
    val nextFireDate: Instant = Instant.now(),
    val lastFiredDate: Instant? = null,
    val timezone: String = java.util.TimeZone.getDefault().id,
    val timezoneMode: TimezoneMode = TimezoneMode.FLOATING,
    val isEnabled: Boolean = true,
    val snoozeCount: Int = 0,
    val persistenceLevel: PersistenceLevel = PersistenceLevel.NOTIFICATION_ONLY,
    val dismissMethod: DismissMethod = DismissMethod.SWIPE,
    val preAlarmMinutes: Int = 0,
    val colorTag: String? = null,
    val iconName: String? = null,
    val syncStatus: SyncStatus = SyncStatus.LOCAL_ONLY,
    val ownerID: String? = null,
    val sharedWith: List<String> = emptyList(),
    val note: String = "",
    val createdAt: Instant = Instant.now(),
    val updatedAt: Instant = Instant.now()
) {
    val isPersistent: Boolean
        get() = persistenceLevel == PersistenceLevel.FULL
}

File: Chronir-iOS/Sources/Core/Models/Alarm.swift
import Foundation

enum CycleType: String, Codable, CaseIterable, Identifiable {
    case daily
    case weekly
    case biweekly
    case monthly
    case quarterly
    case yearly
    case custom

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .daily: return "Daily"
        case .weekly: return "Weekly"
        case .biweekly: return "Biweekly"
        case .monthly: return "Monthly"
        case .quarterly: return "Quarterly"
        case .yearly: return "Yearly"
        case .custom: return "Custom"
        }
    }
}

struct Alarm: Identifiable, Codable, Hashable {
    let id: UUID
    var title: String
    var cycleType: CycleType
    var scheduledTime: Date
    var nextFireDate: Date
    var isEnabled: Bool
    var isPersistent: Bool
    var note: String?
    let createdAt: Date
    var updatedAt: Date

    init(
        id: UUID = UUID(),
        title: String,
        cycleType: CycleType,
        scheduledTime: Date,
        nextFireDate: Date,
        isEnabled: Bool = true,
        isPersistent: Bool = false,
        note: String? = nil,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.title = title
        self.cycleType = cycleType
        self.scheduledTime = scheduledTime
        self.nextFireDate = nextFireDate
        self.isEnabled = isEnabled
        self.isPersistent = isPersistent
        self.note = note
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
